src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -1517,7 +1517,8 @@ codet java_string_library_preprocesst::make_object_get_class_code(
   fun_call.lhs()=class1;
   fun_call.arguments().push_back(string1);
   code_typet fun_type;
-  fun_type.return_type()=string1.type();
+  fun_type.parameters().push_back(code_typet::parametert(string_ptr_type));
+  fun_type.return_type()=class_type;
   fun_call.function().type()=fun_type;
```

Comment:
I'd suggest to re-arrange the code so as to set up that type before creating the `symbol_exprt` above.